### PR TITLE
refactor: use frformats package

### PR DESCRIPTION
**Pull request by Sarrabah**
# Context

The library fr-format has been developed for sharing validation functions between [validata](https://gitlab.com/validata-table) and csv-detective, and to introduce a standard library to validate typical French formats. 

The aim of this PR is to replace custom validation with the implementation of fr-format.

# Refactorings

- code postal et code commune Insee
- code canton, numero departement, code region
- code fantoir
- canton, departement, commune, region, pays

# Behavior changes

- After noticing code Fantoir prefixes starting with two letters, we slightly changed the regex to accept this format (eg. ZB12A).

# Performance

```
Performance report ─=≡Σ((( つ•̀ω•́)つ

Testing table with 100000000 rows

                     without       with     fr-format

"8730"

code_postal          10.27 s       10.62 s
code_fantoir         10.22 s       10.66 s
code_commune         10.27 s       10.47 s

"ABCDE"

code_postal          12.32 s       11.40 s
code_fantoir         12.24 s       11.37 s
code_commune         11.73 s       11.33 s

"12345"

code_postal          11.63 s       11.38 s
code_fantoir         11.23 s       11.05 s
code_commune         11.31 s       10.97 s
```

The differences do not appear to be statistically significant, given the variability between the two executions observed.

### CODE REVIEW

File: csv_detective/detect_fields/FR/geo/code_commune_insee/__init__.py [hunk 1]
```diff
@@ -1,24 +1,5 @@
...
-codes_insee = set(codes_insee)
-f.close()
-# vérification de cohérence avec prise en compte corse 2A/2B et DOM (971-976 sauf 975)
-regex = r'^([01345678][0-9]{4}|2[AB1-9][0-9]{3}|9([0-5][0-9]{3}|7[12346][0-9]{2}))$'
 
-
-def _is(val):
-    '''Renvoie True si val peut être un code commune INSEE, False sinon'''
-    # test sur la longueur
-    if len(val) != 5:
```
Comment (pierrecamilleri) [inline csv_detective/detect_fields/FR/geo/code_commune_insee/__init__.py:18]: Check if this should be kept for performance reasons or not before merge.

File: csv_detective/detect_fields/FR/geo/code_fantoir/__init__.py [hunk 2]
```diff
@@ -1,15 +1,7 @@
...
-
-def _is(val):
-    '''Renvoie True si val peut être un code FANTOIR/RIVOLI, False sinon'''
-    if not bool(re.match(r'^[0-9A-Z][0-9]{3}[ABCDEFGHJKLMNPRSTUVWXYZ]$', val)):
-        return False
-
-    return val[:4] in codes_fantoir
+def _is (val):
+    instance = CodeFantoir()
+    return instance.is_valid(val)
```
Comment (pierrecamilleri) [inline csv_detective/detect_fields/FR/geo/code_fantoir/__init__.py]: `is_valid` is a classmethod and don't need an instanciation

File: tests/test_file.py [hunk 3]
```diff
@@ -35,16 +35,17 @@ def test_columns_output_on_file():
     assert output["columns"]["STRUCTURED_INFO"]["format"] == "json"
     assert output["columns"]["GEO_INFO"]["python_type"] == "json"
     assert output["columns"]["GEO_INFO"]["format"] == "json_geojson"
-    assert output["columns"]["NUMEPCI"]["format"] == "siren"
-
 
+#@pytest.mark.skip(reason="failed for the code_fantoir part")
```
Comment (pierrecamilleri) [inline tests/test_file.py]: Remove comment

File: tests/test_file.py [hunk 4]
```diff
@@ -35,16 +35,17 @@ def test_columns_output_on_file():
...
 def test_profile_output_on_file():
     output = explore_csv.routine(
         csv_file_path="tests/a_test_file.csv",
         num_rows=-1,
         output_profile=True,
         save_results=False,
-    )
+    )    
+    #this test failed when i changed the code_fantoir (WIP)
```
Comment (pierrecamilleri) [inline tests/test_file.py]: Idem

**Review by pierrecamilleri — CHANGES_REQUESTED** [review]

File: csv_detective/detect_fields/FR/geo/code_fantoir/__init__.py (hunk 2 above)
Comment (Sarrabah) [reply csv_detective/detect_fields/FR/geo/code_fantoir/__init__.py]: Done

File: tests/test_file.py (hunk 3 above)
Comment (Sarrabah) [reply tests/test_file.py]: Done

File: tests/test_file.py (hunk 4 above)
Comment (Sarrabah) [reply tests/test_file.py]: Done

File: csv_detective/detect_fields/FR/geo/code_commune_insee/__init__.py (hunk 1 above)
Comment (pierrecamilleri) [reply csv_detective/detect_fields/FR/geo/code_commune_insee/__init__.py:18]: No significant performance change has been noticed (see description).

**Comment by pierrecamilleri** [comment]
Il manque un dernier truc pour faire la PR sur csv detective : 

- faire une release de fr-format
- indiquer fr-format comme dépendance de csv-detective. 

Je m'occupe de ces deux points dès que tu as fait la review de la PR qui améliore la performance de fr-format.CodeFantoir !

**Pull request merged**